### PR TITLE
OPHJOD-856: Update competences -page to new design

### PR DESCRIPTION
**Pull request by henrikuo**
## Description

- Show  new add&edit buttons and arrow icon in profile/competences
- Remove button row from bottom of page
- Fix same translation being used multiple times

## Related JIRA ticket

https://jira.eduuni.fi/browse/OPHJOD-856

NEW:
Screenshot 2024-10-23 at 8 42 19

OLD

Screenshot 2024-10-23 at 8 46 05

### CODE REVIEW

File: src/routes/Profile/Competences/GroupBySource.tsx [hunk 1]
```diff
@@ -1,52 +1,129 @@
...
+                  <Link
+                    to={competenceLinks(competence)}
+                    key={competence}
+                    type="button"
+                    className="text-button-md hover:underline px-5 py-3 text-accent mt-3"
+                  >
+                    <div className="flex flex-row justify-start">
+                      <div className="flex items-center gap-3">
+                        {t('profile.competences.edit')}
+                        <ArrowIcon />
```
Comment (sauanto) [inline src/routes/Profile/Competences/GroupBySource.tsx:103]:
```suggestion
                        <MdArrowForward size={24} />
```

File: src/routes/Profile/Competences/GroupBySource.tsx [hunk 2]
```diff
@@ -1,52 +1,129 @@
...
+                      </div>
+                    </div>
+                  </Link>
+                </div>
+              ) : (
+                <Link
+                  to={competenceLinks(competence)}
+                  key={competence}
+                  type="button"
+                  className="text-button-md hover:underline px-5 py-3 text-accent"
```
Comment (sauanto) [inline src/routes/Profile/Competences/GroupBySource.tsx:113]:
```suggestion
                  className="text-button-md hover:underline mt-3 text-accent"
```

File: src/routes/Profile/Competences/GroupBySource.tsx [hunk 3]
```diff
@@ -1,52 +1,129 @@
...
+                          />
+                        )
+                      );
+                    })}
+                  </div>
+                  <Link
+                    to={competenceLinks(competence)}
+                    key={competence}
+                    type="button"
+                    className="text-button-md hover:underline px-5 py-3 text-accent mt-3"
```
Comment (sauanto) [inline src/routes/Profile/Competences/GroupBySource.tsx:98]:
```suggestion
                    className="text-button-md hover:underline text-accent mt-5"
```

**Review by sauanto — APPROVED** [review]

File: src/routes/Profile/Competences/constants.ts [hunk 4]
```diff
@@ -14,6 +14,7 @@ export interface FilterData {
 
 export type FiltersType = Record<OsaaminenLahdeTyyppi, FilterData[]>;
 export const FILTERS_ORDER = ['TOIMENKUVA', 'KOULUTUS', 'PATEVYYS', 'KIINNOSTUS', 'MUU_OSAAMINEN'] as const;
```
Comment (ketsappi) [inline src/routes/Profile/Competences/constants.ts:16]: If I understand the use of `FILTERS_ORDERS` it should not have the 'KIINNOSTUS' as one of the values?

File: src/i18n/fi/translation.json [hunk 5]
```diff
@@ -204,7 +201,8 @@
   "privacy-policy": "Tietosuojaseloste",
   "profile": {
     "competences": {
-      "add": "Lisää osaaminen",
+      "add": "Lisää osaamisia",
+      "edit": "Muokkaa osaamisia",
```
Comment (ketsappi) [inline src/i18n/fi/translation.json:205]: Good job of unifying these add/edit translations as they were unwanted duplication. 🚀

File: src/routes/Profile/Competences/constants.ts (hunk 4 above)
Comment (henrikuo) [reply src/routes/Profile/Competences/constants.ts:16]: I think ´KIINNOSTUS´ is used in backend and is used somewhere in the system, but in this case the competences -page uses only these 4 competences.  🤷

File: src/routes/Profile/Competences/GroupBySource.tsx (hunk 1 above)
Comment (henrikuo) [reply src/routes/Profile/Competences/GroupBySource.tsx]: Changed to use icon from react-icons 👍

File: src/routes/Profile/Competences/GroupBySource.tsx (hunk 2 above)
Comment (henrikuo) [reply src/routes/Profile/Competences/GroupBySource.tsx]: Removed paddings and used margin where needed

File: src/routes/Profile/Competences/constants.ts (hunk 4 above)
Comment (ketsappi) [reply src/routes/Profile/Competences/constants.ts:16]: Let's keep this for now and if needed, refactoring this later on a separate ticket.

**Review by ketsappi — APPROVED** [review]

**Pull request merged**